# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessibleBase.cpp
# Commit: 32f37b8b54ab
# Full Hash: 32f37b8b54abe5446842bd93b6c3f8666ca6b518
# Author: Morgan Rae Reschenberg <mreschenberg@mozilla.com>
# Date: 2023-01-05 03:40:49
# Description:
#   Bug 1807639: Null check `ancestor` before creating a Pivot r=eeejay
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D165923
# ==============================================================================

diff -r 1cb90708c2ee -r 32f37b8b54ab accessible/ipc/RemoteAccessibleBase.cpp
--- a/accessible/ipc/RemoteAccessibleBase.cpp	Wed Jan 04 21:25:40 2023 +0000
+++ b/accessible/ipc/RemoteAccessibleBase.cpp	Wed Jan 04 21:33:59 2023 +0000
@@ -753,12 +753,17 @@
       while (ancestor && ancestor->Role() != roles::FORM && ancestor != mDoc) {
         ancestor = ancestor->RemoteParent();
       }
-      Pivot p = Pivot(ancestor);
-      PivotRadioNameRule rule(name);
-      Accessible* match = p.Next(ancestor, rule);
-      while (match) {
-        rel.AppendTarget(match->AsRemote());
-        match = p.Next(match, rule);
+      if (ancestor) {
+        // Sometimes we end up with an unparented acc here, potentially
+        // because the acc is being moved. See bug 1807639.
+        // Pivot expects to be created with a non-null mRoot.
+        Pivot p = Pivot(ancestor);
+        PivotRadioNameRule rule(name);
+        Accessible* match = p.Next(ancestor, rule);
+        while (match) {
+          rel.AppendTarget(match->AsRemote());
+          match = p.Next(match, rule);
+        }
       }
       return rel;
     }
